src/os/ui/ogc/ogcserver.js

```diff
@@ -821,6 +843,64 @@ os.ui.ogc.OGCServer.prototype.displayConsentAlerts = function(result) {
 };
 
 
+/**
+ * Extract relevant information from the GetCapabilities result for display in Descriptors, HoverText, etc
+ * @param {string} constraints
+ * @return {?Array<string>}
+ */
+os.ui.ogc.OGCServer.prototype.getProperties = function(constraints) {
+  var properties = [];
+  var xmlDocument = null;
+
+  if (constraints && constraints.indexOf('xml version') && window.DOMParser) {
+    if (constraints.indexOf('&amp;lt;') >= 0) {
+      constraints = constraints.replace(/&amp;/g, '&');
+    }
+    try {
+      // clean up the HTML-encoded strings, e.g. &gt;
+      xmlDocument = (new DOMParser()).parseFromString(goog.string.unescapeEntities(constraints), 'text/xml');
+    } catch (e) {
+      goog.log.error(os.ui.ogc.OGCServer.LOGGER_, 'Could not parse XML for OGCServer additional information', e);
+    }
+  }
+
+  try {
+    var setting = /** @type {Object<?, osx.ogc.ServerProperty>} */ (os.config.Settings.getInstance()
+        .get(os.ui.ogc.OGCServer.DESCRIPTION_KEY, {}));
+    if (setting) {
+      var configs = Object.entries(setting);
+
+      // Display info
+      configs.forEach(([key, config]) => {
+        // help out the compiler
+        key = /** @type {string} */ (key);
+        config = /** @type {osx.ogc.ServerProperty} */ (config);
```

Comment:
The compiler shouldn't need this help. Breaking out the object into an an array with the above `Object.entries` call also seems unnecessary.